### ISSUE TITLE
Whether I can get the structure of the request parameters and response parameters of the grpc interface, this is very useful to me

### ISSUE THREAD

**Issue by l54808821** [fullstorydev/grpcurl#425]
Whether I can get the structure of the request parameters and response parameters of the grpc interface, this is very useful to me

**Comment by l54808821** [comment]
<img width="1129" alt="image" src="https://github.com/fullstorydev/grpcurl/assets/49773498/6cb8e1f8-7d64-4c11-8d5b-92c792fb79a3">
What I get now is only a simple description, which does not satisfy me

**Comment by jhump** [comment]
@l54808821, the `--protoset-out` flag will save the schema to a file that is a binary encoded [`google.protobuf.FileDescriptorSet`](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/descriptor.proto#L54-L58). That is the typical format of a Protobuf "schema".

This file includes the definitions from the proto source file that defines the elements you are listing or describing as well as all of the files it imports. Other Protobuf runtime APIs that support reflection often have a way to work with these descriptors. In Go, for example, it's various packages under `google.golang.org/protobuf/reflection`, with [`protodesc.NewFiles`](https://pkg.go.dev/google.golang.org/protobuf@v1.31.0/reflect/protodesc#NewFiles) being a useful entry point. The general technique is to read in the file contents, unmarshal those bytes into a `google.protobuf.FileDescriptorSet` message, and then use these other APIs to turn that message into a more useful type.

**Comment by l54808821** [comment]
Hello, "fullstorydev/grpcui" project this interface (" http://127.0.0.1:63806/metadata? method=team.Team.add ") return information is what I want, I wonder if it can be obtained by command in this project
<img width="611" alt="image" src="https://github.com/fullstorydev/grpcurl/assets/49773498/1fe6466d-6e6e-4203-b7a2-afc00701976e">
<img width="982" alt="image" src="https://github.com/fullstorydev/grpcurl/assets/49773498/3154349c-84d7-4ea1-a1f4-3226fdaf0a1d">

**Comment by jhump** [comment]
> I wonder if it can be obtained by command in this project

No, not quite. Most of the logic in this package uses the Go protobuf runtime's reflection facilities: https://pkg.go.dev/google.golang.org/protobuf/reflect

As such, nearly everything works with descriptors (per my earlier reply). Only the browser JS code (which doesn't use a protobuf runtime so doesn't have any way to work with descriptors) uses this representation, for building the HTML form. As such this format is not available outside of the endpoint you highlighted.

What exactly are you trying to do that would make this representation useful?

**Comment by l54808821** [comment]
@jhump I made a web tool that needed to get a preset request body based on the grpc interface, so
